telemetry/spans.go

```diff
@@ -119,6 +120,30 @@ func (c *spanCommonBlock) Bytes() []byte {
 	return buf.Bytes()
 }
 
+// SpanCommonBlockBuilder is a builder for SpanCommonBlock.
+type SpanCommonBlockBuilder struct {
+	commonAttributes map[string]interface{}
+}
+
+// WithAttributes sets the common attributes for the builder.
+func (b *SpanCommonBlockBuilder) WithAttributes(commonAttributes map[string]interface{}) *SpanCommonBlockBuilder {
+	b.commonAttributes = commonAttributes
+	return b
+}
+
+// Build creates a SpanCommonBlock from the builder.
+func (b *SpanCommonBlockBuilder) Build() (*SpanCommonBlock, error) {
```

Comment:
If this returns a `PayloadEntry` instead of `*SpanCommonClock` then SpanCommonBlock won't need to be exported anymore.

Reply:
Good call!